File: src/test/java/codeExamples/libExamples/java/example4/Class1.java
package codeExamples.libExamples.java.example4;

public class Class1 {

    public void f1() {
        System.out.println("Class1.f1() is called");
        System.out.println("Internal call: function Class1.f3()");
        f3();
    }

    public void f2() {
        System.out.println("Class1.f2() is called");
    }

    public static void f3() {
        System.out.println("Class1.f3() is called");
        System.out.println("Internal call: function Class1.f4()");
        f4();
    }

    public static void f4() {
        System.out.println("Class1.f4() is called");
    }
}

File: src/test/kotlin/codeExamples/libExamples/example3/Class1.kt
package codeExamples.libExamples.example3

class Class1 {

    fun f1() {
        println("Class1.f1() is called")
        println("Internal call: function Class1.f3()")
        f3()
    }

    fun f2() {
        println("Class1.f2() is called")
    }

    fun f3() {
        println("Class1.f3() is called")
        println("Internal call: function Class1.f4()")
        f4()
    }

    fun f4() {
        println("Class1.f4() is called")
    }
}